### PR TITLE
Add the documentation for the UndefinedVariable rule (Resolves #693 )

### LINKED ISSUE #693: Missing documentation for Undefined Variable Rule
## Description
While creating a changelog for 2.8.0 #692 @kylekatarnls found that we miss documentation for the new Undefined Variable Rule (Added with #497 )
This should be added and is a good issue for people who want to start.

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] Tell if you have the possibility to create a Pull Request to solve this issue.
* [x] If possible add a URL to the incorrect website page or documentation file.
* [x] This issue is about 1 documentation or website issue and nothing more.

### PR DESCRIPTION
**Pull request by tvbeek**
Type: documentation update
Issue: Resolves #693 
Breaking change: no

This PR adds the documentation for the UndefinedVariable rule (Added with #497 )

### CODE REVIEW

File: src/site/rst/rules/cleancode.rst [hunk 1]
```diff
@@ -118,6 +118,20 @@ Example: ::
       return new \stdClass();
   }
 
+UndefinedVariable
+=============
+
+Since: PHPMD 2.8.0
+
+Detects when a variable is used that has not been defined before.
```
Comment (kylekatarnls) [inline src/site/rst/rules/cleancode.rst]: I think the right order is:

> Detects when a variable that is used has not been defined before.

File: src/site/rst/rules/cleancode.rst [hunk 2]
```diff
@@ -118,6 +118,20 @@ Example: ::
       return new \stdClass();
   }
 
+UndefinedVariable
+=============
```
Comment (kylekatarnls) [inline src/site/rst/rules/cleancode.rst]: Length of underline should match title length.

File: src/site/rst/rules/cleancode.rst (hunk 1 above)
Comment (tvbeek) [reply src/site/rst/rules/cleancode.rst]: Changed

File: src/site/rst/rules/cleancode.rst (hunk 2 above)
Comment (tvbeek) [reply src/site/rst/rules/cleancode.rst]: Changed

**Pull request merged**